File: app/src/main/java/com/pythonide/utils/HistoryManager.java
package com.pythonide.utils;

import android.content.Context;
import android.content.SharedPreferences;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.pythonide.models.PythonScript;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class HistoryManager {
    private static final String PREF_NAME = "PythonScriptHistory";
    private static final String KEY_HISTORY = "script_history";
    private static final int MAX_HISTORY_SIZE = 10;
    
    private SharedPreferences prefs;
    private Gson gson;
    
    public HistoryManager(Context context) {
        prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        gson = new Gson();
    }
    
    public void addScript(PythonScript script) {
        List<PythonScript> history = getHistory();
        history.add(0, script); // Add to beginning
        
        // Keep only the last 10 items
        if (history.size() > MAX_HISTORY_SIZE) {
            history = history.subList(0, MAX_HISTORY_SIZE);
        }
        
        // Save updated history
        String json = gson.toJson(history);
        prefs.edit().putString(KEY_HISTORY, json).apply();
    }
    
    public List<PythonScript> getHistory() {
        String json = prefs.getString(KEY_HISTORY, null);
        if (json == null) {
            return new ArrayList<>();
        }
        
        Type type = new TypeToken<List<PythonScript>>(){}.getType();
        List<PythonScript> history = gson.fromJson(json, type);
        return history != null ? history : new ArrayList<>();
    }
    
    public void clearHistory() {
        prefs.edit().remove(KEY_HISTORY).apply();
    }
}

File: app/src/main/java/com/pythonide/utils/HistoryManager.kt
package com.pythonide.utils

import android.content.Context
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class HistoryManager(context: Context) {
    private val prefs = context.getSharedPreferences("PythonIDEHistory", Context.MODE_PRIVATE)
    private val gson = Gson()
    private val historyKey = "code_history"
    private val maxHistory = 10

    fun addToHistory(code: String) {
        val history = getHistory().toMutableList()
        history.add(0, code)
        if (history.size > maxHistory) {
            history.removeAt(history.size - 1)
        }
        saveHistory(history)
    }

    fun getHistory(): List<String> {
        val json = prefs.getString(historyKey, null)
        return if (json != null) {
            val type = object : TypeToken<List<String>>() {}.type
            gson.fromJson(json, type)
        } else {
            emptyList()
        }
    }

    private fun saveHistory(history: List<String>) {
        val json = gson.toJson(history)
        prefs.edit().putString(historyKey, json).apply()
    }
}
